src/SigasiProjectCreator/ArgsAndFileParser.py

```diff
@@ -4,37 +4,40 @@
     :license: BSD, see LICENSE for more details.
 """
 import os
-from optparse import OptionParser
+import argparse
 
 
 class ArgsAndFileParser:
     options = None
 
     def __init__(self, usage):
-        self.parser = OptionParser(usage=usage)
-        self.parser.add_option('-l', '--layout', action="store", dest="layout",
-                               choices=['default', 'simulator'], default='default')
-
-    def parse_args(self, args_len, optional_dir=False):
-        (ArgsAndFileParser.options, args) = self.parser.parse_args()
-
-        if len(args) < args_len:
-            self.parser.error("incorrect number of arguments")
-        if optional_dir:
-            self.check_next_arg_is_dir(args, args_len)
+        self.parser = argparse.ArgumentParser(prog='SigasiProjectCreator')
+        self.parser.add_argument('project_name', help='project name')
+        self.parser.add_argument('input_file', help='input file')
+        self.parser.add_argument('destination_folder', help='root folder of created project')
+        self.parser.add_argument('-l', '--layout', action='store', dest='layout',
+                                 choices=['default', 'simulator'], default='default',
+                                 help='Project layout: default (in place) or simulator (one folder per library with linked files)')
+
+    def parse_args(self):
+        args = self.parser.parse_args()
+        if args.project_name is None:
+            self.parser.error("Incorrect number of arguments: project name missing")
+        if args.input_file is None:
+            self.parser.error("Incorrect number of arguments: input file missing")
+        if not os.path.isfile(args.input_file):
+            self.parser.error("Incorrect number of arguments: given input file does not exist")
+        if args.destination_folder is not None:
+            if not os.path.isdir(args.destination_folder):
```

Comment:
Set `type=pathlib.Path` on the `add_argument` and use `args.destination_folder.is_dir()` instead.

Reply:
Code updated